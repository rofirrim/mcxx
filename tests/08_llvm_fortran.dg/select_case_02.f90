! <testinfo>
! test_generator="config/mercurium-fortran-llvm run"
! test_stdout="select_case_02.f90.out"
! </testinfo>
MODULE MOO
    CONTAINS
        SUBROUTINE SUB(X)
            CHARACTER(LEN=10) :: X
            SELECT CASE(X)
                CASE ("HI")
                    PRINT *, "HELLO!"
                CASE ("BYE")
                    PRINT *, "GOOD BYE!"
                CASE DEFAULT
                    PRINT *, "SAY THAT AGAIN"
            END SELECT
        END SUBROUTINE SUB
END MODULE MOO

PROGRAM MAIN
    USE MOO
    IMPLICIT NONE
    CHARACTER(LEN=10) :: C

    C = "HI"
    CALL SUB(C)

    C = "BYE"
    CALL SUB(C)

    C = "FOO"
    CALL SUB(C)
END PROGRAM MAIN
