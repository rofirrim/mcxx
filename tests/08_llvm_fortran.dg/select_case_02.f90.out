 HELLO!
 GOOD BYE!
 SAY THAT AGAIN
